### PR TITLE
Implement MapDiff class for map difference operations

### PR DESCRIPTION
**Pull request by alimansour0002**
This PR introduces a new `MapDiff` class that allows computing the difference between two maps. The implementation follows the functional programming principles of Cactoos.
Features
Computes the difference between two maps (entries in the first map that are not in the second map)
Supports various input types:
Maps
Iterables of `Map.Entry`
Iterators of `Map.Entry`
Scalar suppliers of `Iterables`
Implementation Details
`MapDiff` extends `MapEnvelope` to maintain the immutability principle
The difference is computed by removing all keys from the first map that exist in the second map
Comprehensive test coverage ensures correct behavior with different input types and edge cases
Testing
The implementation includes tests for:
Basic map difference computation
Empty map handling (both as first and second argument)
Working with standard Java Map implementations
Working with `Iterables` and `Iterators`
This implementation will be useful for functional operations on maps where finding differences between collections is needed.

### CODE REVIEW

**Comment by alimansour0002** [comment]
@yegor256

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Map difference.
+ *
+ * <p>There is no thread-safety guarantee.</p>
+ *
+ * @param <K> Type of key
+ * @param <V> Type of value
+ * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:20]: @alimansour0002 this is not accurate, better say `@since 0.58.0`

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ */
+public final class MapDiff<K, V> extends MapEnvelope<K, V> {
+
+    /**
+     * Ctor.
+     * @param first First map
+     * @param second Second map
+     * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:28]: @alimansour0002 this line is not necessary

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            )
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param first First iterator
+     * @param second Second iterator
+     * @since 1.0
+     */
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:45]: @alimansour0002 this line is not necessary

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                mapFromEntries(second)
+            )
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param first First iterator
+     * @param second Second iterator
+     * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:44]: @alimansour0002 this line is not necessary

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            new IterableOf<>(first),
+            new IterableOf<>(second)
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param first First iterable supplier
+     * @param second Second iterable supplier
+     * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:58]: @alimansour0002 this line is not necessary

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            new Unchecked<>(first).value(),
+            new Unchecked<>(second).value()
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param first First map
+     * @param second Second map
+     * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:74]: @alimansour0002 this line is not necessary

File: src/main/java/org/cactoos/map/MapDiff.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    /**
+     * Compute the difference between two maps.
+     * @param first The first map
+     * @param second The second map
+     * @param <K> Type of keys
+     * @param <V> Type of values
+     * @return The difference map (entries in first but not in second)
+     * @since 1.0
```
Comment (yegor256) [inline src/main/java/org/cactoos/map/MapDiff.java:87]: @alimansour0002 this line is not necessary

File: src/test/java/org/cactoos/map/MapDiffTest.java [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.Map;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasEntry;
+
+/**
+ * Test case for {@link MapDiff}.
+ *
+ * @since 1.0
```
Comment (yegor256) [inline src/test/java/org/cactoos/map/MapDiffTest.java:18]: @alimansour0002 this is not accurate

File: src/test/java/org/cactoos/map/MapDiffTest.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.AvoidDirectAccessToStaticFields"
+    }
+)
+final class MapDiffTest {
+
+    /**
+     * String constant for "one".
+     */
+    private static final String ONE = "one";
```
Comment (yegor256) [inline src/test/java/org/cactoos/map/MapDiffTest.java:31]: @alimansour0002 can't you make each test use its own literals? What's the point of sharing them between tests?

File: src/test/java/org/cactoos/map/MapDiffTest.java [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+ * Test case for {@link MapDiff}.
+ *
+ * @since 1.0
+ */
+@SuppressWarnings(
+    {
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.AvoidDirectAccessToStaticFields"
+    }
+)
```
Comment (yegor256) [inline src/test/java/org/cactoos/map/MapDiffTest.java:25]: @alimansour0002 better get rid of code smells

**Comment by alimansour0002** [comment]
@yegor256 fixed all mentioned issues

**Comment by yegor256** [comment]
@rultor merge
